sdk/core/src/main/java/org/apache/mesos/scheduler/DefaultScheduler.java

```diff
@@ -157,12 +177,11 @@ private void initialize(SchedulerDriver driver) throws InterruptedException {
                 deployPlanManager);
         planCoordinator = new DefaultPlanCoordinator(planManagers, planScheduler);
         planCoordinator.subscribe(this);
-        logger.info("Done initializing.");
+        LOGGER.info("Done initializing.");
     }
 
     private void initializeGlobals(SchedulerDriver driver) {
-        logger.info("Initializing globals...");
-        stateStore = new CuratorStateStore(frameworkName, zkConnectionString);
```

Comment:
This was actually by design.  To avoid split brain problems we only allow work to be done post-registration.  The best guarantee we have that metadata isn't mutated in unexpected ways is to not even construct the `StateStore` until after registration.